docs/migration_guide.md

```diff
@@ -4,28 +4,270 @@
+
+The `build_cxx` function from {py:mod}`torii.tools.cxx` has been removed in favor of the {py:func}`compile_cxx <torii.tools.cxx.compile_cxx>`, which offers more control and capabilities.
+
+```python
+# OLD (<1.0.0)
+from torii.tools.cxx import build_cxx
+
+build_dir, output_file = build_cxx(
+	cxx_sources = {
+		'foo.cc': '/* Source Code for foo */'
+	},
+	output_name  = 'foo',
+	include_dirs = [],
+	macros       = []
+)
+
+# NEW (>=1.0.0)
+from torii.tools.cxx import compile_cxx
+
+output_file = compile_cxx(
+	'foo',
+	build_dir,
+	None,
+	source_listings = {
+		'foo.cc': '/* Source for foo */'
+	}
+)
+
+```
+
+For more information see the full documentation on {py:func}`compile_cxx <torii.tools.cxx.compile_cxx>` for all of it's capabilities and examples of more advanced usage.
+
+The equivalent functionality can be maintained with a little bit of scaffolding:
+
+```python
+def build_cxx(
+	cxx_sources: dict[str, str], output_name: str, include_dirs: list[str], macros: list[str]
+)  -> tuple[TemporaryDirectory[str], str]:
+
+	build_dir  = TemporaryDirectory(prefix = 'torii_cxx_')
+	output_dir = Path(build_dir.name)
+
+	output_file = compile_cxx(
+		output_name, output_dir, None, include_paths = include_dirs, defines = {
+			k: v for (k, v) in (macro.strip().split('=', 2) for macro in macros)
+		},
+		source_listings = cxx_sources
+	)
+
+	return (build_dir, output_file.name)
+```
+
+### `@extend` from `torii.util.decorators`
+
+The `@extend` decorator from the {py:mod}`torii.util.decorators` module has been removed outright. There is no replacement for this as it involved injecting things into other classes which breaks the type system and in general was a bad idea.
+
+### `@memoize` from `torii.util.decorators`
+
+The `@memoize` decorator from {py:mod}`torii.util.decorators` has been removed and should be replaced with {py:class}`functools.cache`.
+
+```python
+# OLD (<1.0.0)
+from torii.util.decorators import memoize
+
+@memoize
+def foo(bar: int, baz: int) -> int:
+	...
+
+# NEW (>=1.0.0)
+from functools import cache
+
+@cache
+def foo(bar: int, baz: int) -> int:
+	...
+```
+
+## Library Changes
+
+### `torii.lib.soc.memory`
+
+The contents of the `torii.lib.soc.memory` module have been moved to {py:mod}`torii.lib.mem`, as they are generally useful for non-SoC designs as well as SoC designs.
+
+More specifically the {py:class}`MemoryMap <torii.lib.mem.map.MemoryMap>`, {py:class}`ResourceInfo <torii.lib.mem.map.ResourceInfo>`, and {py:class}`_RangeMap <torii.lib.mem.map._RangeMap>` have been moved to {py:mod}`torii.lib.mem.map`.
+
+```python
+# OLD (<1.0.0)
+from torii.lib.soc.memory import MemoryMap, ResourceInfo
+
+# NEW (>=1.0.0)
+from torii.lib.mem.map import MemoryMap, ResourceInfo
+```
+
+### `torii.lib.soc.wishbone`
+
+The `wishbone` module has been removed from {py:mod}`torii.lib.soc` and moved to {py:mod}`torii.lib.soc.wishbone`, as it is generally useful for non-SoC designs as well as SoC designs.
```

Comment:
`and it is not the only bus type supported in designs`
`torii.lib.soc.wishbone` => `torii.lib.bus.wishbone`